core/include/esp/iomux.h

```diff
@@ -43,10 +43,17 @@ inline static esp_reg_t gpio_iomux_reg(const uint8_t gpio_number)
     return &(IOMUX.PIN[gpio_to_iomux(gpio_number)]);
 }
 
+/**
+ * Set the I/O Mux function. The iomux_num is an IOMUX register number, see
+ * gpio_iomux_reg above to obtain the IOMUX register number of a GPIO number.
+ * The func is an IOMUX_GPIO<n>_FUNC_<function> value, see iomux_regs.h
+ *
+ * ie *gpio_iomux_reg(3) is equivalent to IOMUX_GPIO3
+ */
 inline static void iomux_set_function(uint8_t iomux_num, uint32_t func)
 {
     uint32_t prev = IOMUX.PIN[iomux_num] & ~IOMUX_PIN_FUNC_MASK;
-    IOMUX.PIN[iomux_num] = IOMUX_FUNC(func) | prev;
+    IOMUX.PIN[iomux_num] = func | prev;
```

Comment:
If you remove `IOMUX_FUNC` here it should be added in function `iomux_set_gpio_function`, otherwise `iomux_set_gpio_function` will be broken.